File: Editorial/293_Flip_Game/CPP/1_Approach_Linear_Iteration.cpp
class Solution {
public:
    vector<string> generatePossibleNextMoves(string currentState) {
        // Array to store all possible next states after making one move.
        vector<string> nextPossibleStates;

        // Iterate through the 'currentState' string characters from left to right.
        for (int index = 0; index < currentState.size() - 1; ++index) {
            // If two adjacent characters of the 'currentState' string are '+', 
            // replace them with '-' and store the new state string.
            if (currentState[index] == '+' and currentState[index + 1] == '+') {
                string nextState = (
                    currentState.substr(0, index) + 
                    "--" + 
                    currentState.substr(index + 2)
                );
                nextPossibleStates.push_back(nextState);
            }
        }
        // Return the 'nextPossibleStates' array.
        return nextPossibleStates;
    }
};

File: Editorial/293_Flip_Game/Go/1_Approach_Linear_Iteration.go
func generatePossibleNextMoves(currentState string) []string {
    // Slice to store all possible next states after making one move.
	var nextPossibleStates []string

	// Iterate through the 'currentState' string characters from left to right.
	for index := 0; index < len(currentState)-1; index++ {
		// If two adjacent characters of the 'currentState' string are '+', 
		// replace them with '-' and store the new state string.
		if currentState[index] == '+' && currentState[index+1] == '+' {
			nextState := currentState[:index] + "--" + currentState[index+2:]
			nextPossibleStates = append(nextPossibleStates, nextState)
		}
	}

	// Return 'nextPossibleStates' slice.
	return nextPossibleStates
}
